### PR TITLE
fix: add warning about python sdk login_user breaking change in 21.4.0

### PR DESCRIPTION
**Pull request by joeldodge79**

### CODE REVIEW

File: packages/sdk-codegen/src/python.gen.ts [hunk 1]
```diff
@@ -401,15 +402,20 @@ ${this.hooks.join('\n')}
...
     const args = this.httpArgs(bump, method)
-    const methodCall = `${indent}response = ${this.it(
+    let methodCall = `${indent}response = ${this.it(
       method.httpMethod.toLowerCase()
     )}`
+    if (method.name === 'login_user') {
+      methodCall =
+        `${indent}warnings.warn("login_user behavior changed significantly ` +
+        `in 21.4.0. See https://git.io/JOtH1")\n${methodCall}`
+    }
```
Comment (jkaster) [inline packages/sdk-codegen/src/python.gen.ts:418]: Want to create an issue to remove this after 3 - 6 months?

**Review by jkaster — APPROVED** [review]
LGTM except for failing CI checks

**Review by jkaster — APPROVED** [review]
LGTM

**Review by jkaster — APPROVED** [review]

**Review by josephaxisa — COMMENTED** [review]
The Python part lgtm. Maybe there was a conversation I missed, but why is the example miner change part of this PR?

File: .github/workflows/codegen-ci.yml [hunk 2]
```diff
@@ -23,8 +23,10 @@ on:
 
 env:
   LOOKERSDK_BASE_URL: https://localhost:20000
-  LOOKERSDK_VERIFY_SSL: false
+  LOOKERSDK_VERIFY_SSL: False
```
Comment (jkaster) [inline .github/workflows/codegen-ci.yml:26]: case sensitivity for the ... win?

File: .github/workflows/codegen-ci.yml [hunk 3]
```diff
@@ -66,10 +62,9 @@ jobs:
       - name: Repo Checkout
         uses: actions/checkout@v2
 
-      - name: Install Node ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
+      - uses: actions/setup-node@v1
         with:
-          node-version: ${{ matrix.node-version }}
+          node-version: 15.x
```
Comment (jkaster) [inline .github/workflows/codegen-ci.yml:67]: that'll speed things up

File: .github/workflows/codegen-ci.yml [hunk 4]
```diff
@@ -80,15 +75,36 @@ jobs:
...
-          echo "base_url=https://self-signed.looker.com:19999" >> looker.ini
-          echo "client_id=id" >> looker.ini
-          echo "client_secret=secret" >> looker.ini
-          echo "verify_ssl=False" >> looker.ini
-          echo "timeout=31" >> looker.ini
-          echo "[Looker31]" >> looker.ini
-          echo "base_url=https://self-signed.looker.com:19999" >> looker.ini
-          echo "verify_ssl=False" >> looker.ini
+          echo "base_url=https://localhost:20000" >> looker.ini
+          echo "verify_ssl=false" >> looker.ini
```
Comment (jkaster) [inline .github/workflows/codegen-ci.yml:79]: our SDKs **should** treat `false` and `False` identically FWIW

File: packages/sdk-codegen-scripts/src/exampleMiner.ts [hunk 5]
```diff
@@ -152,9 +152,17 @@ export const getRemoteOrigin = () => {
...
+  const gitExtractor = /git@github\.com:(.*)\.git/gi
+  let match = gitExtractor.exec(origin)
+  if (!match) {
+    // git origin on CI: https://github.com/looker-open-source/sdk-codegen
+    const httpExtractor = /(https:\/\/github.com.*)(|.git)/
+    match = httpExtractor.exec(origin)
+    if (!match) {
+      return ''
+    }
+    return match[1]
```
Comment (jkaster) [inline packages/sdk-codegen-scripts/src/exampleMiner.ts:164]: I bet this was fun to track down

File: packages/sdk-codegen-scripts/src/fetchSpec.ts [hunk 6]
```diff
@@ -112,8 +112,8 @@ export const logout = async (props: ISDKConfigProps, token: string) => {
 export const login = async (props: ISDKConfigProps) => {
   const xp = specTransport(props)
   const creds = {
-    client_id: props.client_id,
-    client_secret: props.client_secret,
+    client_id: props.client_id ?? process.env.LOOKERSDK_CLIENT_ID,
+    client_secret: props.client_secret ?? process.env.LOOKERSDK_CLIENT_SECRET,
```
Comment (jkaster) [inline packages/sdk-codegen-scripts/src/fetchSpec.ts:116]: Need to fix that config resolution downstream in the future

**Review by jkaster — APPROVED** [review]
Nice detective work

**Review by jkaster — APPROVED** [review]

File: .github/workflows/codegen-ci.yml (hunk 4 above)
Comment (joeldodge79) [reply .github/workflows/codegen-ci.yml]: I could almost swear that the path it takes in `fetchSpec` (where you just `ApiConfig` it) it was failing with `False` and then passed with `false`.

**Pull request merged**